src/graph/data_apis.cc

```diff
@@ -0,0 +1,60 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file graph/graph.cc
+ * \brief DGL graph index APIs
+ */
+
+#include <string.h>
+#include <dgl/graph.h>
+#include "../c_api_common.h"
+
+using dgl::runtime::DGLArgs;
+using dgl::runtime::DGLArgValue;
+using dgl::runtime::DGLRetValue;
+using dgl::runtime::PackedFunc;
+using dgl::runtime::NDArray;
+
+namespace dgl {
+
+std::unordered_map<std::string, DLDataType> dtype_map = {
+  {"float32", DLDataType{kDLFloat, 32, 1}},
+  {"int32", DLDataType{kDLInt, 32, 1}},
+};
+
+DGL_REGISTER_GLOBAL("contrib.graph_store._CAPI_DGLAllocateSharedMem")
+.set_body([] (DGLArgs args, DGLRetValue* rv) {
+    std::string mem_name = args[0];
+    dgl::runtime::NDArray shape_arr = args[1];
+    std::string dtype_str = args[2];
+    std::string fill = args[3];
+    bool is_create = args[4];
+    auto it = dtype_map.find(dtype_str);
```

Comment:
For dtype, we shall use the type defined [here](https://github.com/dmlc/dgl/blob/master/python/dgl/_ffi/runtime_ctypes.py#L36). To pass it to CAPI, checkout [`DGLArrayAlloc`](https://github.com/dmlc/dgl/blob/cd4f0a2bfce4b58c4f5006d9760eb446df05289b/include/dgl/runtime/c_runtime_api.h#L373).

Reply:
actually, the data type string should be consistent with https://github.com/dmlc/dgl/blob/master/python/dgl/backend/pytorch/tensor.py#L10
The data type is extract from the backend tensor.